abe/abe.go

```diff
@@ -0,0 +1,464 @@
+package abe
+
+import (
+	"fmt"
+	"math/big"
+
+	"strconv"
+	"strings"
+
+	"github.com/cloudflare/bn256"
+	"github.com/fentec-project/gofe/data"
+	"github.com/fentec-project/gofe/sample"
+)
+
+// abeParams represents configuration parameters for the ABE scheme instance.
+type abeParams struct {
+	L int      // number of attributes
+	P *big.Int // order of the elliptic curve
+}
+
+// Abe represents an ABE scheme.
+type ABE struct {
+	Params *abeParams
+}
+
+// NewAbe configures a new instance of the scheme.
+// It accepts l the number of attributes possibly used in
+// the scheme. Attributes' names will be considered as
+// elements of a set {0, 1,..., l-1}.
+func NewAbe(l int) *ABE {
```

Comment:
Rename to `NewABE` to be consistent - sorry, I meant to state this in my last comment.